MSVC/gen-build-config.bat

```diff
@@ -49,79 +50,91 @@ SET $BUILDCONFIG_TEMPLATE=build_config.h.in
 SET $BUILDCONFIG_FILE=build_config.h
 SET $VERSIONINFO_FILE=MSVC\versioninfo.h
 
-SET $USE_SQLITE3 0
-SET $USE_POSTGRESQL 0
-SET $USE_LEVELDB 0
-SET $USE_REDIS 0
-SET $USE_ICONV 0
-SET $PACKAGING_FLAT 1
-SET $INSTALL_PREFIX ""
+SET $USE_SQLITE3=0
+SET $USE_POSTGRESQL=0
+SET $USE_LEVELDB=0
+SET $USE_REDIS=0
+SET $USE_ICONV=0
+SET $PACKAGING_FLAT=1
+SET $INSTALL_PREFIX=\\
+SET $PROJECTDIR=""
 
-SET $USE_ANY 0
-SET $VERBOSE 0
-SET $TEST 0
+
+SET $USE_ANY=0
+SET $VERBOSE=0
+SET $TEST=0
 
 :: Command-line argument parsing
 :PARSEARGS
-	IF /I "%~1" == "-h" (
-		CALL USAGE
+	IF /I "%~1" =="-h" (
+		CALL :USAGE
 		EXIT /B 0
-	) ELSE IF /I "%~1" == "--help" (
-		CALL USAGE
+	)
+	IF /I "%~1" =="--help" (
+		CALL :USAGE
 		EXIT /B 0
-	) ELSE IF /I "%~1" == "-v" (
-		SET $VERBOSE 1
-	) ELSE IF /I "%~1" == "--verbose" (
-		SET $VERBOSE 1
-	) ELSE IF /I "%~1" == "--test" (
-		SET $TEST 1
-	) ELSE IF /I "%~1" == "--use-sqlite3" (
-		SET $USE_SQLITE3 1
-		SET $USE_ANY 1
-	) ELSE IF /I "%~1" == "--use-postgresql" (
-		SET $USE_POSTGRESQL 1
-		SET $USE_ANY 1
-	) ELSE IF /I "%~1" == "--use-leveldb" (
-		SET $USE_LEVELDB 1
-		SET $USE_ANY 1
-	) ELSE IF /I "%~1" == "--use-redis" (
-		SET $USE_REDIS 1
-		SET $USE_ANY 1
-	) ELSE (
+	)
+	IF /I "%~1" =="-v" (
+		SET $VERBOSE=1
+	)
+	IF /I "%~1" =="--verbose" (
+		SET $VERBOSE=1
+	)
+	IF /I "%~1" =="--test" (
+		SET $TEST=1
+	)
+	IF /I "%~1" =="--use-sqlite3" (
+		SET $USE_SQLITE3=1
+		SET $USE_ANY=1
+	)
+	IF /I "%~1" =="--use-postgresql" (
+		SET $USE_POSTGRESQL=1
+		SET $USE_ANY=1
+	)
+	IF /I "%~1" =="--use-leveldb" (
+		SET $USE_LEVELDB=1
+		SET $USE_ANY=1
+	)
+	IF /I "%~1" =="--use-redis" (
+		SET $USE_REDIS=1
+		SET $USE_ANY=1
+	)
+	IF /I "%~1" =="-p" (
+		SET $PROJECTDIR=%~f2
+		:: SHIFT twice, because we took 2 arguments here.
+		SHIFT
+	)
+	IF /I "%~1" =="" (
 		GOTO GEN_CONFIG
 	)
 	SHIFT
 	GOTO PARSEARGS
 
 :GEN_CONFIG
-
-IF %$USE_ANY% NEQ 1 (
+IF NOT %$USE_ANY% ==1 (
 	ECHO %$MYNAME%: Error: at least one --use-^<database^> option must be given
-	CALL USAGE
+	CALL :USAGE
 	EXIT /B 1
 )
-
-IF "%~1" == "" (
+IF %$PROJECTDIR% =="" (
 	ECHO %$MYNAME%: Error: PROJECTDIR argument is required
-	CALL USAGE
+	CALL :USAGE
 	EXIT /B 1
-) ELSE (
-	SET $PROJECTDIR=%~f1
 )
 
-CALL CHECK_FOR_GIT
-IF ERRORLEVEL 1 (
-	CALL GET_NONGIT_VERSION
-) ELSE
-	CALL GET_GIT_VERSION
+CALL :CHECK_FOR_GIT
+IF ERRORLEVEL==1 (
```

Comment:
Oh yes, you are right, the Syntax is `ERRORLEVEL <number>` but it works because `ERRORLEVEL==1` gets translated to `ERRORLEVEL 1`.
ERRORLEVEL 1.

I changed it accidently, Ill revert it.